File: engine/src/main/java/com/terralite/engine/simulation/WorldSimulationAdapter.java
package com.terralite.engine.simulation;

import com.terralite.engine.world.World;

import java.util.Objects;

public final class WorldSimulationAdapter implements SimulationSystem {
    private final World world;
    private final WorldSimulationSystem system;

    public WorldSimulationAdapter(World world, WorldSimulationSystem system) {
        this.world = Objects.requireNonNull(world, "world");
        this.system = Objects.requireNonNull(system, "system");
    }

    @Override
    public void tick(SimulationTick tick) {
        system.tick(world, tick);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/simulation/WorldSimulationAdapter.kt
package com.terralite.engine.simulation

import com.terralite.engine.world.World

class WorldSimulationAdapter(
    private val world: World,
    private val system: WorldSimulationSystem
) : SimulationSystem {
    override fun tick(tick: SimulationTick) {
        system.tick(world, tick)
    }
}
